src/validation/validation.cpp

```diff
@@ -3706,7 +3706,7 @@ bool ProcessNewBlock(CValidationState &state,
         // demerit the sender
         return error("%s: CheckBlockHeader FAILED", __func__);
     }
-    if (IsChainNearlySyncd() && !fImporting && !fReindex)
+    if (IsChainNearlySyncd() && !fImporting && !fReindex && connmgr->ExpeditedBlockNodes().size())
         SendExpeditedBlock(*pblock, pfrom);
```

Comment:
I think it needs to fall through to sendexpedited so that the right sequence of locks can be taken.  Also you would need a lock to get  the ExpeditedBlockNodes.size() but then I don't think your lockorder wouldn't be correct anymore.

Reply:
As of now the chain of actions goes like this:

0) the if statement this PR change
1) `SendExpeditedBlock` (lock `cs_main`)
2a) lock `cs_expedited`
2b) `IsRecentlyExpeditedAndStore` (`assertlockheld(cs_expedited)`)
3) `ActuallySendExpeditedBlock` (no lock)

with this changes we are just stopping at 0 in case we don't have any Xpedited capable nodes.

That said `ExpeditedBlockNodes()` is defined as the following

```c++
VNodeRefs CConnMgr::ExpeditedBlockNodes`()
{
    LOCK(cs_expedited);

    VNodeRefs vRefs;

    for (CNode *pNode : vExpeditedUpstream)
        vRefs.push_back(CNodeRef(pNode));

    return vRefs;
}
```

which means that the lock on the needed data structure has been taken care of, unless I missing something obvious.

Reply:
yes, you are correct, I just assumed you were grabbing the size from common container.